lib/masamune/abstract_syntax_tree/comment.rb

```diff
@@ -0,0 +1,23 @@
+# TODO: This has only been tested with one-line comments.
+
+module Masamune
+  class AbstractSyntaxTree
+    class Comment < Prism::Comment
+      attr_reader :ast_id
+
+      include NodeHelper
+
+      def initialize(comment_type, comment_location, ast_id)
+        @ast_id = ast_id
+        super(comment_type, comment_location)
+      end
+
+      # We grab the comment from the original Prism object's source because
+      # Prism::Comment doesn't have a method that shows the string itself.
```

Comment:
Comments have locations, which are the same locations as on the nodes. So you can call `comment.location.slice` to get the original value.

Reply:
I see, I was under the impression that because we couldn't get it with `comment.slice` (like regular nodes) that `comment.location.slice` wouldn't exist either. I switched to using the `comments` array from the `Prism::ParseResult` object as you suggested: https://github.com/gazayas/masamune-ast/pull/60/commits/4c6e9b188bdee6e15d17d032812fb7c5d902e40f

I did notice that it doesn't have `\n` at the end like Ripper lex output does:
```ruby
comments = <<~CODE
  # First comment
  # Second comment
CODE

p Prism::parse(comment).comments.first.location.slice
#=> "# First comment"
```

```
[1] pry(#<TestMasamune>)> Ripper.lex(comments)
=> [[[1, 0], :on_comment, "# First comment\n", BEG], [[2, 0], :on_comment, "# Second comment\n", BEG]]
```

This isn't too big of an issue for us right now though, so I'll stick to using the `comments` array.